cpp/src/visitors/bfs_visitor.cu

```diff
@@ -0,0 +1,138 @@
+/*
```

Comment:
Any chance this could be a .cpp file?  Think about what it might take to accomplish that.  I don't see any direct thrust calls here.

Reply:
I will look into this.

Reply:
Done.